### PR TITLE
Generalize student t

### PR DESCRIPTION
**Pull request by t4c1**
## Summary

Generalizes student t distribution related functions to accept Eigen expressions.

## Tests

Tested using expression testing framework.

## Side Effects

None.

## Release notes

Generalized student t distribution related functions to accept Eigen expressions.

## Checklist

- [ ] Math issue #1470 

- [ ] Copyright holder: Tadej Ciglarič

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [ ] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [ ] the code is written in idiomatic C++ and changes are documented in the doxygen

- [ ] the new changes are tested

### CODE REVIEW

File: test/expressions/generateExpressionTests.py [hunk 1]
```diff
@@ -176,12 +176,15 @@ def parse_signature(signature):
     "lkj_corr_lpdf": [1, None],
     "lkj_corr_log": [1, None],
     "log_diff_exp": [3, None],
-    "log_inv_logit_diff": [1.2, 0.4],
+    "log_inv_logit_diff": [1.2, 0.4],
```
Comment (bbbales2) [inline test/expressions/generateExpressionTests.py:179]: Ooo friendly neighborhood space

File: test/expressions/generateExpressionTests.py [hunk 2]
```diff
@@ -176,12 +176,15 @@ def parse_signature(signature):
...
     "log_diff_exp": [3, None],
-    "log_inv_logit_diff": [1.2, 0.4],
+    "log_inv_logit_diff": [1.2, 0.4],  
     "multinomial_log" : [None, 1],
     "multinomial_lpmf" : [None, 1],
     "multinomial_rng" : [1, None, None],
-    "unit_vector_free" : [1.0],
-
+    "unit_vector_free" : [1.0],    "student_t_cdf" : [0.8, None, 0.4, None],
+    "student_t_cdf_log" : [0.8, None, 0.4, None],
```
Comment (bbbales2) [inline test/expressions/generateExpressionTests.py:184]: For student_t arguments 2 and 4 need to be positive, but the others are unconstrained (https://mc-stan.org/docs/2_24/functions-reference/student-t-distribution.html). Why the constraint here?

File: test/expressions/generateExpressionTests.py [hunk 3]
```diff
@@ -237,12 +240,12 @@ def save_tests_in_files(N_files, tests):
 
 def handle_function_list(functions_input, signatures):
     """
-    Handles list of functions, splitting elements between functions and signatures.
+    Handles list of functions, splitting elements between functions and signatures.
```
Comment (bbbales2) [inline test/expressions/generateExpressionTests.py:243]: Remove spaces if you end up changing anything in this pull, otherwise leave them.

**Review by bbbales2 — DISMISSED** [review]
Good, merge ~~on the 20th~~, after the feature freeze

File: test/expressions/generateExpressionTests.py (hunk 2 above)
Comment (t4c1) [reply test/expressions/generateExpressionTests.py:184]: By default all arguments have same values. With y==mu I was getting NaNs.

Comment (bbbales2) [reply test/expressions/generateExpressionTests.py:184]: Got it. This is fine then.

**Comment by bbbales2** [comment]
Conflicts

File: test/expressions/generateExpressionTests.py (hunk 3 above)
Comment (t4c1) [reply test/expressions/generateExpressionTests.py]: fixed

File: test/expressions/generateExpressionTests.py (hunk 1 above)
Comment (t4c1) [reply test/expressions/generateExpressionTests.py]: removed

**Review by bbbales2 — APPROVED** [review]
Good!

**Pull request merged**